### PR TITLE
feat: detailed status messages

### PR DESCRIPTION
**Pull request by olevski**
Provide by-container summary of the status.

### CODE REVIEW

File: controller/server_status.py [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+        for status in self.statuses:
+            if status.failed:
+                failed_statuses.append(status)
+            elif status.running_ready or status.completed_successfully:
+                ok_statuses.append(status)
+        for status in self.init_statuses:
+            if status.failed:
+                failed_init_statuses.append(status)
+            elif status.running_ready or status.completed_successfully:
+                ok_init_statuses.append(status)
```
Comment (Panaetius) [inline controller/server_status.py:301]: The list here is not actually used, it could just be ints that get incremented. Also, could `self.statuses` and `self.init_statuses` be chained/combined and just a single for loop used? And then below it'd just be something like

`and num_ok_statuses == len(self.statuses)+len(self.init_statuses)` ? Actually you could just count total number of statuses in the for loop and not use `len` at all.

**Review by Panaetius — CHANGES_REQUESTED** [review]
Thank you, looks good. The logic for the different statuses is a bit hard to follow in places, but then I'm not sure how it could be simplified/more self-explanatory.

There's a typo in a comment and once suggestion that you can ignore.

File: controller/server_status.py (hunk 1 above)
Comment (olevski) [reply controller/server_status.py]: @Panaetius I think I originally had some other plans with those and that is why I had the arrays. Anyhow you are right all that stuff is not needed.

**Review by Panaetius — APPROVED** [review]
